connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -473,25 +474,30 @@ export async function confluenceUpdatePagesParentIdsActivity(
 
   // Utilize an in-memory map to cache page hierarchies, thereby reducing database queries.
   const cachedHierarchy = await getSpaceHierarchy(connectorId, spaceId);
-  for (const page of pages) {
-    // Retrieve parents using the internal ID, which aligns with the permissions
-    // view rendering and RAG requirements.
-    const parentIds = await getConfluencePageParentIds(
-      connectorId,
-      page,
-      cachedHierarchy
-    );
 
-    await updateDocumentParentsField({
-      dataSourceConfig: {
-        dataSourceName: connector.dataSourceName,
-        workspaceId: connector.workspaceId,
-        workspaceAPIKey: connector.workspaceAPIKey,
-      },
-      documentId: makeConfluencePageId(page.pageId),
-      parents: parentIds,
-    });
-  }
+  await concurrentExecutor(
+    pages,
+    async (page) => {
+      // Retrieve parents using the internal ID, which aligns with the permissions
+      // view rendering and RAG requirements.
+      const parentIds = await getConfluencePageParentIds(
```

Comment:
I'll do a follow up PR to change the types here to use a readonly map to make it more explicit that this is thread safe.